cmake/distcc.cmake

```diff
@@ -0,0 +1,21 @@
+option(ENABLE_DISTCC "Distribute build with distcc." OFF)
+if(ENABLE_DISTCC)
+  find_program (DISTCC_FOUND distcc)
+  if(DISTCC_FOUND)
+    message(STATUS "Building with distcc: ${DISTCC_FOUND}")
+
+    if ("${CMAKE_C_COMPILER_LAUNCHER}" STREQUAL "")
+      set(CMAKE_C_COMPILER_LAUNCHER "distcc")
+    else()
+      set(CMAKE_C_COMPILER_LAUNCHER "${CMAKE_C_COMPILER_LAUNCHER};distcc")
```

Comment:
nit, since `<LANG>_COMPILER_LAUNCHER` is a semicolon-separated list. could just use something like
```cmake
list(APPEND CMAKE_C_COMPILER_LAUNCHER "distcc")
```
or better off
```cmake
list(APPEND CMAKE_C_COMPILER_LAUNCHER "${DISTCC_EXECUTABLE}")
```

Reply:
also, because we don't have any .c files in the project by itself. all of the .c source files could be compiled as part of the building process come from `cmake/oss.cmake.in`, if we want to use distcc to compile them as well, shall we populate the `CMAKE_C_COMPILER_LAUNCHER` option and build flags and other settings down to https://github.com/redpanda-data/redpanda/blob/884086e90dcc9790e4df8c9f549bfd4c4f9ec423/cmake/main.cmake#L95-L97 ?

Reply:
> nit, since `<LANG>_COMPILER_LAUNCHER` is a semicolon-separated list. could just use something like
> 
> ```cmake
> list(APPEND CMAKE_C_COMPILER_LAUNCHER "distcc")
> ```
> 
> or better off
> 
> ```cmake
> list(APPEND CMAKE_C_COMPILER_LAUNCHER "${DISTCC_EXECUTABLE}")
> ```

nice, thanks for the suggestion, fixed.



> also, because we don't have any .c files in the project by itself. all of the .c source files could be compiled as part of the building process come from `cmake/oss.cmake.in`, if we want to use distcc to compile them as well, shall we populate the `CMAKE_C_COMPILER_LAUNCHER` option and build flags and other settings down to
> 
> https://github.com/redpanda-data/redpanda/blob/884086e90dcc9790e4df8c9f549bfd4c4f9ec423/cmake/main.cmake#L95-L97
> 
> ?

I see in oss.cmake.in a set of common cmake args is built which inherits the launcher options [here](https://github.com/redpanda-data/redpanda/blob/681995973107bd9d0d25b98be6c41e9d8d32cc00/cmake/oss.cmake.in#L19), so it should automatically pick up the values (for those deps which do use `common_cmake_args`) ?

Reply:
ahh, indeed! sorry i missed the `common_cmake_args` option passed to the external projects.